addons/mail/models/discuss/discuss_channel_category.py

```diff
@@ -0,0 +1,30 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class ChannelCategory(models.Model):
+    _name = "discuss.channel.category"
+    _description = "Channel Category"
+
+    name = fields.Char('Channel category name', required=True)
+    channel_ids = fields.One2many('discuss.channel', 'channel_category_id', string='Channels')
+
+    @api.model
+    def get_categories(self):
+        categories = self.env['discuss.channel.category'].search([])
+        return Store(categories).get_result()
+
+    def _to_store(self, store: Store):
+        """Adds channel categories data to the given store."""
+        fields = {
+            "name": True,
+        }
+        for category in self:
+            data = category._read_format(
+                list(fields),
+                load=False,
+            )[0]
+            data["threads"] = Store.many(category.channel_ids)
```

Comment:
Note that this field is probably a bad idea regardless.
Unbound many fields will run into scaling issues, like we already have currently.
We should never return all the channels of a category, but only a subset.
I'd suggest this `_to_store` should only return the name.
Each channels store should return `channel_category_id` instead.